docs/boilerplate/app/utils/useAppTheme.tsx.md

```diff
@@ -0,0 +1,131 @@
+  backgroundColor: theme.colors.background,
+  // Accessing the color palette is not generally recommended.
+  color: theme.colors.palette.angry500,
+  // Spacing can be changed on a per-theme basis as well.
+  paddingHorizontal: theme.spacing.small,
+})
+
+// We haven't abandoned plain style objects as the preferred way
+// to style your components, but it can't use dynamic themes.
+const $plainObjectStyle: ViewStyle = {
+  marginBottom: 20,
+}
+```
+
+## Properties:
+
+### `navTheme`
+
+A `react-navigtion` [theme object](https://reactnavigation.org/docs/themes#built-in-themes). This is the same object you would pass to a `NavigationContainer` component.
+
+### `setThemeContextOverride`
+
+A function that allows you to override the theme context. This is useful for allowing users to switch between light and dark mode.
+
+The default behavior is to use the system theme, but you can override this by calling `setThemeContextOverride("light" | "dark")`.
+
+Calling `setThemeContextOverride(undefined)` will reset the theme to the user's system preference.
+
+### `theme`
+
+A `Theme` object that contains all the colors, spacing, and other theme-related properties of the current theme context.
+
+### `themeContext`
+
+A string that represents the current theme context. This will almost always be either "light" or "dark".
+
+:::info
+
+When resetting the theme context to the system preference with `setThemeContextOverride(undefined)`, the `themeContext` will not be undefined. It will be the return value of `react-native`'s `useColorScheme()` hook.
+
+:::
+
+### `themed`
+
+`themed()` is a simple function with complex types. You can pass a plain style object to it, a `ThemedStyle` function, or an array of either.
+
+:::note
+
+When passing an array of styles or `ThemedStyle` functions to `themed()`, it will return a single style object with last properties overriding any properties previously set.
+
+In the example below, the last `<View>` would be yellow even though `$themedStyle` specifies red because the background color property was modified by a later style in the array.
+
+:::
+
+```tsx
+const $plainStyle: ViewStyle = {
+  padding: 10,
+  backgroundColor: "black",
+  width: 25,
+  height: 25,
+}
+
+const $themedStyle: ThemedStyle<ViewStyle> = (theme) => ({
+  backgroundColor: theme.colors.errorBackground,
+  height: theme.spacing.xl,
+  width: theme.spacing.xl
+})
+
+<View style={$plainStyle} />
+// <View style={$themedStyle} /> // This won't work
+<View style={themed($themedStyle)} />
+// <View style={themed($plainStyle)} /> // You can, but why would you?
+<View style={themed([$themedStyle, $plainStyle])} />
+<View style={themed([
+  $themedStyle,
+  $plainStyle,
+  // A plain style object
+  { width: 7, backgroundColor: 'yellow'}
+])} />
+```
+
+Make sure you don't pass any `Animated` styles to `themed()`. It will not work as expected! Keep them in separate style array objects: `<Animatd.View style={[$animatedStyle, themed($myThemedStyle)]}>`.
```

Comment:
```suggestion
Make sure you don't pass any `Animated` styles to `themed()`. It will not work as expected! Keep them in separate style array objects: `<Animated.View style={[$animatedStyle, themed($myThemedStyle)]}>`.
```